### PR TITLE
Upgrade to the yocto scarthgap

### PR DESCRIPTION
**Pull request by rshym**
- snort: add packages for snort-tsn?
- yaml: add usrmerge to the DISTRO_FEATURES
- yaml: move systemd to the common configuration
- HACK: util-linux: remove libmount-mountfd-support
- domu: add meta-networking layer
- yaml: add .rootfs to all image names
- Upgrade to the yocto scarthgap

### CODE REVIEW

**Comment by rshym** [comment]
Tested on Spider 1.2 with Dom0+DomD+DomU.
Works as expected.
On hold till merged related PRs for xt-common and xt-rcar.

**Comment by lorc** [comment]
I am not sure that all the changes should be in the same patch. For example, how CMA is realated to snort packages?

File: layers/meta-xt-domx-gen4/recipes-core/util-linux/util-linux_2.%.bbappend [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This is temporary fix required util vendor provides linux 6.1+
+# because required API is not present in linux 5.10.
+#
+# See detailed explanation in the poky's commit
+# b5f4d8952a7e3ea22672951db470f87c65bc8821
+# "util-linux: Add PACKAGECONFIG option to mitigate rootfs remount error"
+PACKAGECONFIG:remove = "libmount-mountfd-support"
```
Comment (GrygiriiS) [inline layers/meta-xt-domx-gen4/recipes-core/util-linux/util-linux_2.%.bbappend:7]: this is a hack - separate patch before updating LAYERSERIES_COMPAT_meta* and yaml branches.

File: layers/meta-xt-dom0-gen4/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 2]
```diff
@@ -756,7 +756,7 @@ CONFIG_TRANSPARENT_HUGEPAGE_ALWAYS=y
 # CONFIG_TRANSPARENT_HUGEPAGE_MADVISE is not set
 # CONFIG_CLEANCACHE is not set
 # CONFIG_FRONTSWAP is not set
-CONFIG_CMA=y
+# CONFIG_CMA is not set
```
Comment (GrygiriiS) [inline layers/meta-xt-dom0-gen4/recipes-kernel/linux/linux-generic-armv8/defconfig:759]: Separate patch - hack/tmp to w/a binary size issue.

File: prod-devel-rcar-s4.yaml [hunk 3]
```diff
@@ -61,7 +61,7 @@ common_data:
     - [SOC_FAMILY, "%{SOC_FAMILY}"]
 
     # Add systemd configuration
-    - [DISTRO_FEATURES:append, " systemd"]
+    - [DISTRO_FEATURES:append, " usrmerge systemd"]
```
Comment (GrygiriiS) [inline prod-devel-rcar-s4.yaml:64]: what is usrmerge? Why?

File: prod-devel-rcar-s4.yaml [hunk 4]
```diff
@@ -180,7 +180,7 @@ components:
         - "../../layers/meta-xt-driver-domain-gen4"
       target_images:
         - "tmp/deploy/images/%{MACHINE}/Image"
-        - "tmp/deploy/images/%{MACHINE}/rcar-image-minimal-%{MACHINE}.ext4"
+        - "tmp/deploy/images/%{MACHINE}/rcar-image-minimal-%{MACHINE}.rootfs.ext4"
```
Comment (GrygiriiS) [inline prod-devel-rcar-s4.yaml:183]: Why?

File: prod-devel-rcar-s4.yaml [hunk 5]
```diff
@@ -210,6 +210,7 @@ components:
 
       layers:
         - "../meta-openembedded/meta-oe"
+        - "../meta-openembedded/meta-networking"
```
Comment (GrygiriiS) [inline prod-devel-rcar-s4.yaml:213]: Why?

**Comment by GrygiriiS** [comment]
`Reviewed-by: Grygorii Strashko <grygorii_strashko@epam.com>`

File: layers/meta-xt-domx-gen4/recipes-core/util-linux/util-linux_2.%.bbappend (hunk 1 above)
Comment (rshym) [reply layers/meta-xt-domx-gen4/recipes-core/util-linux/util-linux_2.%.bbappend:7]: Separate commit provided.

File: layers/meta-xt-dom0-gen4/recipes-kernel/linux/linux-generic-armv8/defconfig (hunk 2 above)
Comment (rshym) [reply layers/meta-xt-dom0-gen4/recipes-kernel/linux/linux-generic-armv8/defconfig]: Separate PR made and already merged.

File: prod-devel-rcar-s4.yaml (hunk 4 above)
Comment (rshym) [reply prod-devel-rcar-s4.yaml:180]: Explanation is provided.

File: prod-devel-rcar-s4.yaml (hunk 5 above)
Comment (rshym) [reply prod-devel-rcar-s4.yaml]: Explanation is provided.

**Comment by rshym** [comment]
> I am not sure that all the changes should be in the same patch. For example, how CMA is realated to snort packages?

The original patch is splitted, and explanations are provided.

**Comment by rshym** [comment]
`Rb` is added.
Tested.
@lorc ready to be merged.

**Pull request merged**